src/content/docs/apm/agents/go-agent/instrumentation/go-agent-code-level-metrics-instrument.mdx

```diff
@@ -0,0 +1,124 @@
+defer txn.End()
+```
+
+### Explicitly Mark the Code Location Based On a Function Value [#instrument-transactions-by-function-clm]
+If the code you wish to instrument is available to you as a `func` type value, such as the name of a function or a function literal value, you can specify that as the location for code-level metrics reporting by adding a `WithFunctionLocation` option to `StartTransaction`, passing the `func` value as a parameter:
+```go
+func myfunction() { ... }
+.
+.
+.
+txn := app.StartTransaction("myfunction", newrelic.WithFunctionLocation(myfunction))
+```
+or
+```go
+someFunction := func() {...}
+.
+.
+.
+txn := app.StartTransaction("myfunction", newrelic.WithFunctionLocation(someFunction))
+```
+
+You may also obtain a `CodeLocation` value from a `func` value for later use with the `WithCodeLocation` option. Compare this to the usage [shown above](#instrument-transactions-any-location-clm) to save a code location with `ThisCodeLocation` for later reference with `WithCodeLocation`. This time we do the same thing but with a `func` value instead:
+```go
+func myFunc() {...}
+.
+.
+.
+locationOfMyFunc, err := newrelic.FunctionLocation(myFunc)
+if err != nil {
+    // handle the case that the location could not be determined
+    // from the value passed to FunctionLocation.
+}
+.
+.
+.
+txn := app.StartTransaction("mytransaction", newrelic.WithCodeLocation(locationOfMyFunc))
+```
+Note that `FunctionLocation` returns an error if the value passed to it was not a valid function, or the source code location could not be obtained from it. By contrast, the `WithFunctionLocation` option sets the code-level metrics based on the value passed if possible, but silently does nothing if an error occurs.
+
+## Adding Custom Options to Wrapped Handlers [#instrument-transactions-custom-wrappers-clm]
+The same options described above which may be added to the end of `StartTransaction` may also be added to `WrapHandle` and `WrapHandleFunc` to
+adjust the code-level metrics collection associated with transactions started by them, if desired (although in most cases the `WrapHandle` and `WrapHandleFunc` functions will correctly identify the location of the code being instrumented). For example:
+```go
+http.HandleFunc(newrelic.WrapHandleFunc(app, "/endpoint", endpointFunction, newrelic.WithThisCodeLocation()))
+```
+
+## Changing Trace Options After a Transaction Has Started [#instrument-transactions-set-options-clm]
+Sometimes you don't have the luxury of knowing the code location until after the transaction has already been started (perhaps by a framework or integration function on your behalf). You may change the transaction options on an existing transaction by calling its `SetOption` method. The parameters are a set of `TraceOption` functions just as described above. For example:
+```go
+txn := newrelic.FromContext(r.context)
+txn.SetOption(newrelic.WithThisCodeLocation())
+```
```

Comment:
I'd like a closing paragraph or a wrap up. Ending on code makes the doc feel unfinished in my opinion.

Reply:
@homelessbirds some of the previous sections end with code snippets and don't have a summary of the section. Let's chat about this for the next big revision.